services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/UpdateItemOperationTest.java

```diff
@@ -164,34 +157,24 @@ public void generateRequest_withConditionExpression() {
         FakeItemWithSort item = createUniqueFakeItemWithSort();
         item.setOtherAttribute1("value-1");
         UpdateItemOperation<FakeItemWithSort> updateItemOperation =
-            UpdateItemOperation.create(UpdateItemEnhancedRequest.builder(FakeItemWithSort.class)
-                                                                .item(item)
-                                                                .conditionExpression(CONDITION_EXPRESSION)
-                                                                .build());
-        Map<String, AttributeValue> expectedKey = new HashMap<>();
-        expectedKey.put("id", AttributeValue.builder().s(item.getId()).build());
-        expectedKey.put("sort", AttributeValue.builder().s(item.getSort()).build());
+            UpdateItemOperation.create(requestFakeItemWithSort(item, b -> b.conditionExpression(CONDITION_EXPRESSION)));
+
+        UpdateItemRequest request = updateItemOperation.generateRequest(FakeItemWithSort.getTableSchema(), PRIMARY_CONTEXT, null);
+
+
+        String expectedUpdateExpression = "SET " + OTHER_ATTRIBUTE_1_NAME + " = " + OTHER_ATTRIBUTE_1_VALUE +
+                                          " REMOVE " + OTHER_ATTRIBUTE_2_NAME;
         Map<String, AttributeValue> expectedValues = new HashMap<>(CONDITION_EXPRESSION.expressionValues());
         expectedValues.put(OTHER_ATTRIBUTE_1_VALUE, AttributeValue.builder().s("value-1").build());
         Map<String, String> expectedNames = new HashMap<>(CONDITION_EXPRESSION.expressionNames());
         expectedNames.put(OTHER_ATTRIBUTE_1_NAME, "other_attribute_1");
         expectedNames.put(OTHER_ATTRIBUTE_2_NAME, "other_attribute_2");
-        UpdateItemRequest.Builder baseExpectedRequest =
-            UpdateItemRequest.builder()
-                             .tableName(TABLE_NAME)
-                             .expressionAttributeValues(expectedValues)
-                             .expressionAttributeNames(expectedNames)
-                             .conditionExpression(CONDITION_EXPRESSION.expression())
-                             .key(expectedKey)
-                             .returnValues(ReturnValue.ALL_NEW);
-        UpdateItemRequest expectedRequest =
-            baseExpectedRequest.updateExpression("SET " + OTHER_ATTRIBUTE_1_NAME + " = " + OTHER_ATTRIBUTE_1_VALUE +
-                                                 " REMOVE " + OTHER_ATTRIBUTE_2_NAME)
-                               .build();
-
-        UpdateItemRequest request = updateItemOperation.generateRequest(FakeItemWithSort.getTableSchema(),
-                                                                        PRIMARY_CONTEXT,
-                                                                        null);
+        UpdateItemRequest.Builder expectedRequestBuilder = ddbRequestBuilder(ddbKey(item.getId(), item.getSort()));
+        expectedRequestBuilder.expressionAttributeValues(expectedValues);
```

Comment:
Nit: I think it helps readability to chain the builder method calls together, rather than repeatedly reference `expectedRequestBuilder`. Same feedback for other locations.

Reply:
Agreed... I stopped refactoring these types of tests after a while because I'll be adding quite a few similar ones, and I didn't want to end up having to back out of some optimization. I would like to simplify the handling of the updateExpression/expressionNames/expressionMaps/conditionals to a point where it's easy to glance at a test and quickly see the significant bits. Expect to see something in the next PR.